src/functions/reportGen.ts

```diff
@@ -23,33 +22,27 @@ const reportGen: Handler = async (event: any, context?: Context, callback?: Call
   }
   const lambdaService = new LambdaService(new LambdaClient({}));
   const reportService: ReportGenerationService = new ReportGenerationService(new TestResultsService(lambdaService), new ActivitiesService(lambdaService));
-  const atfReportPromises: Promise<PutObjectRequest>[] = [];
-
   const sendATFReport: SendATFReport = new SendATFReport();
 
-  event.Records.forEach((record: any) => {
-    const recordBody = JSON.parse(record?.body)
-    const visit: any = unmarshall(recordBody?.dynamodb.NewImage);
-
-    if (visit) {
-      const atfReportPromise = reportService
-        .generateATFReport(visit)
-        .then((generationServiceResponse) => {
-          return sendATFReport.sendATFReport(generationServiceResponse, visit);
-        })
-        .catch((error: any) => {
-          console.log(error);
-          throw error;
-        });
+  console.debug("Services injected, looping over sqs events");
+  try {
+    event.Records.forEach(async (record: any) => {
```

Comment:
Probably would be better to do a for of loop here as the promises aren't being used

Reply:
Yes, consistent with the approach in Notification Service looks better IMO